### PR TITLE
Ported rq job function to its own module, in order to be able to easi…

### PR DESCRIPTION
**Pull request by spyrostz**
…ly lazy-import packages only in case they are needed (essentially only when a job has really started). That way the main worker process that forks processes that execute the simulation jobs will not grow its memory utilization.

### CODE REVIEW

File: src/d3a/d3a_core/d3a_jobs.py [hunk 1]
```diff
@@ -15,139 +15,21 @@
...
-        d3a.constants.IS_CANARY_NETWORK = scenario.pop("is_canary_network", False)
-        d3a.constants.RUN_IN_REALTIME = d3a.constants.IS_CANARY_NETWORK
-    saved_state = decompress_and_decode_queued_strings(saved_state)
-    log.error(f"Scenario: {scenario}")
-    log.error(f"Settings: {settings}")
-    log.error(f"Events: {events}")
-    log.error(f"Aggregator device mapping: {aggregator_device_mapping}")
-    log.error(f"Previous simulation state: {saved_state}")
-
+    from d3a.d3a_core.rq_job_handler import launch_simulation_from_rq_job
```
Comment (faizan2590) [inline src/d3a/d3a_core/d3a_jobs.py:28]: I assume this is remnant of testing and lazy import could be avoided here

File: src/d3a/d3a_core/rq_job_handler.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        d3a.constants.COLLABORATION_ID = scenario.pop("collaboration_uuid")
+        d3a.constants.EXTERNAL_CONNECTION_WEB = True
+        d3a.constants.IS_CANARY_NETWORK = scenario.pop("is_canary_network", False)
+        d3a.constants.RUN_IN_REALTIME = d3a.constants.IS_CANARY_NETWORK
+    saved_state = decompress_and_decode_queued_strings(saved_state)
+    log.error(f"Scenario: {scenario}")
+    log.error(f"Settings: {settings}")
+    log.error(f"Events: {events}")
+    log.error(f"Aggregator device mapping: {aggregator_device_mapping}")
+    log.error(f"Previous simulation state: {saved_state}")
```
Comment (faizan2590) [inline src/d3a/d3a_core/rq_job_handler.py:38]: Cant these all be squashed into single error log?

File: src/d3a/d3a_core/rq_job_handler.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        run_simulation(setup_module_name=scenario_name,
+                       simulation_config=config,
+                       simulation_events=events,
+                       redis_job_id=job_id,
+                       saved_sim_state=saved_state,
+                       slot_length_realtime=slot_length_realtime,
+                       kwargs=kwargs)
+    except Exception:
+        import traceback
+        from d3a.d3a_core.redis_connections.redis_communication import publish_job_error_output
```
Comment (faizan2590) [inline src/d3a/d3a_core/rq_job_handler.py:116]: assuming these lazy could be avoided too.

**Review by faizan2590 — DISMISSED** [review]
Some comments, other than that LGTM

**Review by hannesdiedrich — DISMISSED** [review]
LGTM

**Review by mclrch — APPROVED** [review]
LGTM

**Review by Hassan754 — DISMISSED** [review]

File: src/d3a/d3a_core/rq_job_handler.py (hunk 2 above)
Comment (spyrostz) [reply src/d3a/d3a_core/rq_job_handler.py:38]: Sure, merging these, thanks.

**Review by faizan2590 — APPROVED** [review]
LGTM

File: src/d3a/d3a_core/rq_job_handler.py (hunk 2 above)
Comment (hannesdiedrich) [reply src/d3a/d3a_core/rq_job_handler.py]: I am voting for leaving these as is. Each of them are huge for complex sims. So either you add some `\n` in there leave these in order to separate the information. 
I hope you agree.

Comment (hannesdiedrich) [reply src/d3a/d3a_core/rq_job_handler.py]: Ah just saw your commit. All good.

**Pull request merged**